Grammar/python.gram

```diff
@@ -778,6 +780,13 @@ invalid_kwarg:
     | expression a='=' {
         RAISE_SYNTAX_ERROR_KNOWN_LOCATION(
             a, "expression cannot contain assignment, perhaps you meant \"==\"?") }
+
+invalid_expression:
+    # !(NAME STRING) is not matched so we don't show this error with some invalid string prefixes like: kf"dsfsdf"
+    # Soft keywords need to also be ignored because they can be parsed as NAME NAME
+    | !(NAME STRING | SOFT_KEYWORD) a=expression expression {
+        RAISE_ERROR_KNOWN_LOCATION(p, PyExc_SyntaxError, a->lineno, a->end_col_offset - 1,
+            "invalid syntax. Perhaps you forgot a comma?") }
```

Comment:
Why can't we use the `KNOWN_LOCATION` variant here?

Reply:
Oh, I revealed a bit of my next secret mini project too early 😉. Will fix this to use the macro

Reply:
The reason is because I want to point to the end of the expression, not to the begging of it (because that's where the comma should be).

(I deleted my previous comment because I forgot that this was the real reason)

Reply:
Correct.

(Now you've gotten me really interested though.)

Reply:
@lysnikolaou 🤫🤫🤫🤫🤫

```
>>> f(a, b, x for x in range(10))
  File "<stdin>", line 1
    f(a, b, x for x in range(10))
            ^^^^^^^^^^^^^^^^^^^^
SyntaxError: Generator expression must be parenthesized
```

Reply:
@lysnikolaou Actually, I may need help with this if we want to get it to beta freeze. Do you want to give me a hand?

Reply:
Oh wow! Yup, count me in! How should we go about it?

Reply:
Checkout my prototype: https://github.com/pablogsal/cpython/tree/tokenizer_better_errors

This basically adds a "end column offset" to syntax errors and threads it down everywhere.

The moderate bigger task is to make the parser/grammar to populate that accordingly. Now I am doing this hack:

https://github.com/pablogsal/cpython/blob/tokenizer_better_errors/Parser/pegen.c#L426

as a way to populate that.

Basically what's left is to productionalise this. There are several things to do:

* The hack doesn't work always (check one of the new dict errors like `{1:2, 3, 3:3}`.
* Devise a better way to make the parser inject this automatically and maybe a new macro so you can mark the beginning and the end passing two tokens instead of 1.
* Tests